src/halmos/config.py

```diff
@@ -0,0 +1,610 @@
+        global_default=False,
+        group=debugging,
+    )
+
+    print_steps: bool = arg(
+        help="print every execution step",
+        global_default=False,
+        group=debugging,
+    )
+
+    print_mem: bool = arg(
+        help="when --print-steps is enabled, also print memory contents",
+        global_default=False,
+        group=debugging,
+    )
+
+    print_states: bool = arg(
+        help="print all final execution states",
+        global_default=False,
+        group=debugging,
+    )
+
+    print_failed_states: bool = arg(
+        help="print failed execution states",
+        global_default=False,
+        group=debugging,
+    )
+
+    print_blocked_states: bool = arg(
+        help="print blocked execution states",
+        global_default=False,
+        group=debugging,
+    )
+
+    print_setup_states: bool = arg(
+        help="print setup execution states",
+        global_default=False,
+        group=debugging,
+    )
+
+    print_full_model: bool = arg(
+        help="print full counterexample model",
+        global_default=False,
+        group=debugging,
+    )
+
+    early_exit: bool = arg(
+        help="stop after a counterexample is found",
+        global_default=False,
+        group=debugging,
+    )
+
+    dump_smt_queries: bool = arg(
+        help="dump SMT queries for assertion violations",
+        global_default=False,
+        group=debugging,
+    )
+
+    ### Build options
+
+    forge_build_out: str = arg(
+        help="forge build artifacts directory name (default: 'out/')",
+        metavar="DIRECTORY_NAME",
+        global_default="out",
+        group=build,
+    )
+
+    ### Solver options
+
+    smt_exp_by_const: int = arg(
+        help="interpret constant power up to N",
+        global_default=2,
+        metavar="N",
+        group=solver,
+    )
+
+    solver_timeout_branching: int = arg(
+        help="set timeout (in milliseconds) for solving branching conditions; 0 means no timeout",
+        global_default=1,
+        metavar="TIMEOUT",
+        group=solver,
+    )
+
+    solver_timeout_assertion: int = arg(
+        help="set timeout (in milliseconds) for solving assertion violation conditions; 0 means no timeout",
+        global_default=1000,
+        metavar="TIMEOUT",
+        group=solver,
+    )
+
+    solver_max_memory: int = arg(
+        help="set memory limit (in megabytes) for the solver; 0 means no limit",
+        global_default=0,
+        metavar="SIZE",
+        group=solver,
+    )
+
+    solver_command: str = arg(
+        help="use the given command when invoking the solver, e.g. `z3 -model`",
+        global_default=None,
+        metavar="COMMAND",
+        group=solver,
+    )
+
+    solver_parallel: bool = arg(
+        help="run assertion solvers in parallel", global_default=False, group=solver
+    )
+
+    solver_threads: int = arg(
+        help="set the number of threads for parallel solvers",
+        global_default=os.cpu_count() or 1,
```

Comment:
i don't remember why i put the previous default value `min(32, os.cpu_count() + 4)`, but this new one makes sense.

Reply:
yes I was always wondering why I was running with 24 threads when my machine supports 20 😅